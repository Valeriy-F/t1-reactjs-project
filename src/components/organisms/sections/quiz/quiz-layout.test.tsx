import { render, screen } from "@testing-library/react";

import QuizLayout from "./quiz-layout";

const categories = ["Category 1", "Category 2", "Category 3"];

const data = {
  title: "We will select the perfect product for you",
  subtitle: "Answer three questions and we will send you a catalog with the most suitable products for you.",
  bodyTitle: "What type of product are you considering?",
  actionsButtonsData: [
    {
      type: "submit" as const,
      form: "quiz-form",
      title: "Next step",
      onClick: () => {
        // Go to next page
      },
      disabled: false,
    },
  ],
  paginationData: { currentPage: 1, totalPages: 2 },
};

describe("QuizLayout tests", () => {
  test("QuizLayout snapshots", () => {
    render(
      <QuizLayout {...data}>
        {categories.map((category) => (
          <div key={category}>{category}</div>
        ))}
      </QuizLayout>
    );

    const quizComponent = screen.queryByTestId("quiz-layout");

    expect(quizComponent).toBeInTheDocument();
    expect(quizComponent).toMatchSnapshot();
  });
});
