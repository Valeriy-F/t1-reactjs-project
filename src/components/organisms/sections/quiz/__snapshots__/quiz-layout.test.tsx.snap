// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizLayout tests QuizLayout snapshots 1`] = `
<div
  class="content-container"
  data-testid="quiz-layout"
>
  <div
    class="header"
  >
    <div>
      <h2
        class="variant-h2 color-primary"
      >
        We will select the perfect product for you
      </h2>
    </div>
    <div
      class="subtitle"
    >
      <p
        class="variant-text_bold color-primary-light"
      >
        Answer three questions and we will send you a catalog with the most suitable products for you.
      </p>
    </div>
    <div
      class="border color-primary-light thickness-1"
    />
  </div>
  <div
    class="body"
  >
    <div
      class="body__title"
    >
      <h3
        class="variant-h3 color-primary"
      >
        What type of product are you considering?
      </h3>
    </div>
    <div>
      Category 1
    </div>
    <div>
      Category 2
    </div>
    <div>
      Category 3
    </div>
  </div>
  <div
    class="footer"
  >
    <div
      class="footer__border-top"
    >
      <div
        class="border color-primary-light thickness-1"
      />
    </div>
    <div
      class="footer__content"
    >
      <div>
        <p
          class="variant-text_bold color-primary-light"
        >
          1 of 2
        </p>
      </div>
      <div
        class="footer__actions"
      >
        <button
          class="base variant-button size-md color-primary-transparent "
          form="quiz-form"
          type="submit"
        >
          Next step
        </button>
      </div>
    </div>
  </div>
</div>
`;
